sdcm/sct_events/base.py

```diff
@@ -95,13 +95,14 @@ def add_event(self, event: ContinuousEvent):
         if not issubclass(type(event), ContinuousEvent):
             raise ContinuousEventRegistryException(f"Event: {event} is not a ContinuousEvent")
 
+        if self._find_event_by_id(event.event_id):
+            raise ContinuousEventRegistryException("Event %s with unit id: %s already present."
+                                                   "Event ids in the registry must be unique.")
+
         self.continuous_events.append(event)
 
     def get_event_by_id(self, event_id: Union[uuid.UUID, str]) -> Optional[ContinuousEvent]:
-        event_filter = self.get_registry_filter()
-        found_events = event_filter \
-            .filter_by_id(event_id=event_id) \
-            .get_filtered()
+        found_events = self._find_event_by_id(event_id)
 
         if len(found_events) > 1:
```

Comment:
If you are doing it on `add_event`, why do you need it here ?
AFAIK there is no other method for event to get into registry

Reply:
Fixed (removed the check)